Basic integration testing

jcpowermac opened the pull request.

jcpowermac commented:
Current problems:
- IP address of `clusterhost`
- Release pipeline sends emails pipeline fails if unable to send.

jcpowermac commented:
@codificat ok I think this is ready to go.

codificat commented on tests/alternative/README.md:
````diff
@@ -0,0 +1,17 @@
+
+cd tests
+bash run.sh
+```
````
I believe this should read:
```
cd jenkins-on-openshift/tests
bash alternative/run.sh
```

codificat commented on tests/tests-openshift-template.yaml:
```diff
@@ -0,0 +1,78 @@
+      type: Docker
+    triggers:
+    - imageChange: {}
+      type: ImageChange
```
I think we want a `ConfigChange` trigger as well... and/or document that a build of ansible should be performed before running the test pipeline

codificat commented on tests/README.md:
```diff
@@ -0,0 +1,21 @@
+#### Requirements
+
+- OpenShift Container Platform v3.7
+- Permissions to use BuildConfig objects
```
Maybe make it explicit that builds should happen on properly subscribed RHEL7 nodes?

codificat commented on tests/README.md:
```diff
@@ -0,0 +1,21 @@
+an account that can create projects.  This will be used to login
+into OpenShift and Ansible.
+
+
```
... and then:

`oc start-build jenkins-on-openshift-tests`

codificat commented on tests/Jenkinsfile:
```diff
@@ -0,0 +1,47 @@
+
+                        try {
+
+                            sh "oc login ${cluster} --username ${env.OCP_USR} --password ${env.OCP_PSW} --insecure-skip-tls-verify"
```
Need to surround the password in quotes

codificat reviewed (changes requested):
This is awesome, thanks!!

A few minor comments but otherwise looks [very] good to me.

jcpowermac committed:
Initial testing

Run testing on Jenkins using an Ansible slave.
The pipeline executes ansible with a default inventory.
Then starts the pipelines.
